lib/prawn/attachment.rb

```diff
@@ -54,10 +52,23 @@ def attach(src, options = {})
 
     private
 
-    def data_and_opts(path, src, options)
+    def file?(path)
+      File.file?(path)
+    rescue ArgumentError
+      false
+    end
+
+    def directory?(path)
+      File.directory?(path)
+    rescue ArgumentError
+      false
+    end
+
+    def data_and_opts(src, options)
```

Comment:
I'd try to avoid using methods that return two values in ruby. It's not very idiomatic, and we be clearer to have two simple methods.

Reply:
Ok, now there are two methods for both variables